src/Compilers/CSharp/Test/Emit3/FlowAnalysis/StructTests.cs

```diff
@@ -219,6 +219,196 @@ public void GenericStructWithPropertyUsingStruct()
+        public void InstanceMemberExplosion_01()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct A<T>
+{
+    A<A<T>> x;
+}
+
+struct B<T>
+{
+    A<B<T>> x;
+}
+
+struct C<T>
+{
+    D<T> x;
+}
+struct D<T>
+{
+    C<D<T>> x;
+}
+";
+            CreateCompilation(program).VerifyDiagnostics(
+                // (4,13): error CS0523: Struct member 'A<T>.x' of type 'A<A<T>>' causes a cycle in the struct layout
+                //     A<A<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("A<T>.x", "A<A<T>>").WithLocation(4, 13),
+                // (14,10): error CS0523: Struct member 'C<T>.x' of type 'D<T>' causes a cycle in the struct layout
+                //     D<T> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("C<T>.x", "D<T>").WithLocation(14, 10),
+                // (18,13): error CS0523: Struct member 'D<T>.x' of type 'C<D<T>>' causes a cycle in the struct layout
+                //     C<D<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("D<T>.x", "C<D<T>>").WithLocation(18, 13)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void InstanceMemberExplosion_02()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct A<T>
+{
+    A<A<T>> x;
+}
+
+struct B<T>
+{
+    A<C<B<T>>> x;
+}
+
+struct C<T>
+{
+}
+";
+            CreateCompilation(program).VerifyDiagnostics(
+                // (4,13): error CS0523: Struct member 'A<T>.x' of type 'A<A<T>>' causes a cycle in the struct layout
+                //     A<A<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("A<T>.x", "A<A<T>>").WithLocation(4, 13)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void InstanceMemberExplosion_03()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct E
+{}
+
+struct X<T>
+{
+    T _t;
+}
+
+struct Y
+{
+    X<Z> xz;
+}
+
+struct Z
+{
+    X<E> xe;
+    X<Y> xy;
+}
+";
+            CreateCompilation(program).VerifyDiagnostics(
+                // (12,10): error CS0523: Struct member 'Y.xz' of type 'X<Z>' causes a cycle in the struct layout
+                //     X<Z> xz;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "xz").WithArguments("Y.xz", "X<Z>").WithLocation(12, 10),
+                // (18,10): error CS0523: Struct member 'Z.xy' of type 'X<Y>' causes a cycle in the struct layout
+                //     X<Y> xy;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "xy").WithArguments("Z.xy", "X<Y>").WithLocation(18, 10)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
```

Comment:
>https://github.com/dotnet/roslyn/issues/66844 [](http://example.com/codeflow?start=19&length=45)

Are these static cases affected by this issue?

Reply:
> Are these static cases affected by this issue?

The scenario is not broken, if that is what you are asking. But I consider the test related to the issue.